parser_library/test/context/instruction_test.cpp

```diff
@@ -0,0 +1,2500 @@
+/*
+ * Copyright (c) 2022 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Broadcom, Inc. - initial API and implementation
+ */
+
+#include "gtest/gtest.h"
+
+#include "../common_testing.h"
+#include "../mock_parse_lib_provider.h"
+#include "context/instruction.h"
+
+// clang-format off
+namespace {
+std::unordered_map<std::string, const std::set<system_architecture>> machine_instruction_compatibility_matrix = {
```

Comment:
I don't see much value in this test, unless this has been constructed manually and independently from the list in `instruction.cpp`.

Simpler test that just deals with the various `system_architecture` combination without involving the whole instruction list seems much more beneficial.

Reply:
Ok